cmd/sourced/compose/workdir/common.go

```diff
@@ -191,15 +194,15 @@ func ListPaths() ([]string, error) {
 	return res, nil
 }
 
-// RemovePath removes working directory by removing required files
+// RemovePath removes working directory by removing required and optional files,
 // and recursively removes directories up to the workdirs root as long as they are empty
 func RemovePath(path string) error {
 	workdirsRoot, err := workdirsPath()
 	if err != nil {
 		return err
 	}
 
-	for _, f := range RequiredFiles {
+	for _, f := range append(RequiredFiles, OptionalFiles...) {
```

Comment:
Given that we're iterating also on `OptionalFiles` I'd expect a different error handling for them. In particular, I'd expect not returning an error, but just continue if an optional file is not found.

Reply:
True, since we're deleting, we can skip missing files.
But we can not skip optional files, because if they're there, and we're pruning, they must be deleted to fully perform the prune.

done by 2205b4d6bc9579754541ab8fd8d10cdaaf0efcde